pkg/kubelet/allocation/state/state_mem.go

```diff
@@ -65,6 +65,17 @@ func (s *stateMemory) GetPodResourceInfoMap() PodResourceInfoMap {
 	return s.podResources.Clone()
 }
 
+func (s *stateMemory) GetPodResourceInfo(podUID types.UID) (PodResourceInfo, bool) {
+	s.RLock()
+	defer s.RUnlock()
+
+	resourceInfo, ok := s.podResources[podUID]
+	if !ok {
+		return PodResourceInfo{}, ok
+	}
+	return resourceInfo, ok
```

Comment:
nit: I think you can simplify this: 

```suggestion
	resourceInfo, ok := s.podResources[podUID]
	return resourceInfo, ok
```

if !ok, `resourceInfo` will be an empty struct anyways